### PR TITLE
feat(lambda): function construct exposes configured timeout

### PR DESCRIPTION
**Pull request by ben-eb**
Supersedes https://github.com/aws/aws-cdk/pull/17000.

I didn't realise that the "allow edits by maintainers" was not supported under organisation accounts, so I've forked under my account instead. Otherwise, these changes are the same as the linked PR.

Thanks! 👍 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda/lib/function.ts [hunk 1]
```diff
@@ -725,6 +731,7 @@ export class Function extends FunctionBase {
     });
 
     this.runtime = props.runtime;
+    this.timeout = props.timeout;
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-lambda/lib/function.ts:734]: Why not reify this to the 3 seconds we promised it would be?

Comment (ben-eb) [reply packages/@aws-cdk/aws-lambda/lib/function.ts:734]: @rix0rrr See previous discussion: https://github.com/aws/aws-cdk/pull/17000#discussion_r739116360

**Review by nija-at — APPROVED** [review]

**Review by nija-at — CHANGES_REQUESTED** [review]
Can you fix the failing build? Looks like the snippet in the README is not compilable.

**Comment by ben-eb** [comment]
@nija-at Sure thing, have fixed. 🙂

**Review by nija-at — APPROVED** [review]

**Pull request merged**